### PR TITLE
cleanup some code, add bounds checks

### PR DESCRIPTION
**Pull request by segfaultdoc**

### CODE REVIEW

File: tip-payment/Anchor.toml [hunk 1]
```diff
@@ -22,7 +22,6 @@ members = [
     "programs/tip-distribution",
     "programs/tip-payment"
 ]
-exclude = ["programs/merkle-tree"]
```
Comment (segfaultdoc) [inline tip-payment/Anchor.toml:25]: @esemeniuc @0xEdgar double checking this is safe to delete? I don't see any usage here or in jito-solana

Comment (esemeniuc) [reply tip-payment/Anchor.toml:25]: yea that looks like its gone

Comment (esemeniuc) [reply tip-payment/Anchor.toml:25]: yea jito-solana has its own copy: https://github.com/jito-foundation/jito-solana/blob/master/merkle-tree/src/merkle_tree.rs

**Review by esemeniuc — APPROVED** [review]
LGTM

**Pull request merged**